neuralmonkey/encoders/clone_encoder.py

```diff
@@ -0,0 +1,61 @@
+# tests: lint, mypy
+
+from typing import Optional, Any
+from typeguard import check_argument_types
+
+from neuralmonkey.encoders.attentive import Attentive
+from neuralmonkey.model.model_part import ModelPart, FeedDict
+from neuralmonkey.logging import log
+from neuralmonkey.dataset import Dataset
+
+
+class CloneEncoder(ModelPart, Attentive):
+    """
+    A class that clones existing encoder to use for different data_id serie
+    """
+
+    def __init__(self,
+                 name: str,
+                 clone_encoder,
+                 data_id: str,
+                 attention_type: Optional[Any] = None,
+                 save_checkpoint: Optional[str] = None,
+                 load_checkpoint: Optional[str] = None) -> None:
+
+        ModelPart.__init__(self, name, save_checkpoint, load_checkpoint)
+        Attentive.__init__(self, attention_type)
+
+        assert check_argument_types()
+
+        self.clone_encoder = clone_encoder
+        self.data_id = data_id
+
+        self.encoded = self.clone_encoder.encoded
+
+        log("Cloned encoder initialized")
+
+    # pylint: disable=protected-access
+    @property
+    def _attention_tensor(self):
+        return self.clone_encoder._attention_tensor
+
+    # pylint: disable=protected-access
+    @property
+    def _attention_mask(self):
+        return self.clone_encoder._input_mask
+
+    @property
+    def vocabulary_size(self):
+        return len(self.clone_encoder.vocabulary)
+
+    def feed_dict(self, dataset: Dataset, train: bool = False) -> FeedDict:
+        # store original data_id from cloned encoder
+        orig_data_id = self.clone_encoder.data_id
+        self.clone_encoder.data_id = self.data_id
```

Comment:
... nevšiml jsem si toho řádku, co to opravuje... Nicméně je to pořád špatně v hrozně moc - 1 ohledech.